pandas/tests/reshape/test_concat.py

```diff
@@ -829,12 +831,98 @@ def test_append_preserve_index_name(self):
         result = df1.append(df2)
         assert result.index.name == 'A'
 
+    indexes_I = [
+        pd.RangeIndex(3),
+        pd.Index([4, 5, 6]),
+        pd.Index([4.5, 5.5, 6.5]),
+        pd.Index(list('abc')),
+        pd.CategoricalIndex('A B C'.split()),
+        pd.DatetimeIndex([dt.datetime(2013, 1, 3, 0, 0),
+                          dt.datetime(2013, 1, 3, 6, 10),
+                          dt.datetime(2013, 1, 3, 7, 12)]),
+    ]
+    indexes_II = indexes_I + [pd.CategoricalIndex('A B C'.split(),
+                                                  ordered=True)]
+    indexes_III = [
+        pd.IntervalIndex.from_breaks([0, 1, 2, 3]),
+        pd.MultiIndex.from_arrays(['A B C'.split(), 'D E F'.split()]),
+    ]
+
+    @pytest.fixture(params=indexes_II)
+    def index_II(self, request):
+        return request.param
+
+    @pytest.fixture(params=indexes_III)
+    def index_III(self, request):
+        return request.param
+
+    def test_append_same_columns_type(self, index_II):
+        # GH18359
+
+        # df wider than ser
+        df = pd.DataFrame([[1, 2, 3], [4, 5, 6]], columns=index_II)
+        ser_index = index_II[:2]
+        ser = pd.Series([7, 8], index=ser_index, name=2)
+        result = df.append(ser)
+        expected = pd.DataFrame([[1., 2., 3.], [4, 5, 6], [7, 8, np.nan]],
+                                index=[0, 1, 2],
+                                columns=index_II)
+        assert_frame_equal(result, expected)
+
+        # ser wider than df
+        ser_index = index_II
+        index_II = index_II[:2]
+        df = pd.DataFrame([[1, 2], [4, 5]], columns=index_II)
+        ser = pd.Series([7, 8, 9], index=ser_index, name=2)
+        result = df.append(ser)
+        expected = pd.DataFrame([[1, 2, np.nan], [4, 5, np.nan], [7, 8, 9]],
+                                index=[0, 1, 2],
+                                columns=ser_index)
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("df_columns, series_index",
```

Comment:
you can just use 2 parameterize here, one for df_columns and one for series_index; you don't need combinations, pytest already does this.

Reply:
I don't get this approach to work, AFAIKS, parametrize doesn't do this for you, and I've googled it without results. Can you give an example?